src/active_set_quadratic.jl

```diff
@@ -0,0 +1,310 @@
+    else
+        return a.λ * b
+    end
+end
+function ActiveSetQuadratic(tuple_values::AbstractVector{Tuple{R,AT}}, A::UniformScaling, b) where {AT,R}
+    return ActiveSetQuadratic(tuple_values, Identity(A.λ), b)
+end
+function ActiveSetQuadratic{AT,R}(tuple_values::AbstractVector{<:Tuple{<:Number,<:Any}}, A::UniformScaling, b) where {AT,R}
+    return ActiveSetQuadratic{AT,R}(tuple_values, Identity(A.λ), b)
+end
+
+# these three functions do not update the active set iterate
+
+function Base.push!(as::ActiveSetQuadratic{AT,R}, (λ, a)) where {AT,R}
+    dot_x = zero(R)
+    dot_A = Vector{R}(undef, length(as))
+    dot_b = fast_dot(as.b, a)
+    Aa = as.A * a
+    @inbounds for i in 1:length(as)
+        dot_A[i] = fast_dot(Aa, as.atoms[i])
+        as.dots_x[i] += λ * dot_A[i]
+        dot_x += as.weights[i] * dot_A[i]
+    end
+    push!(dot_A, fast_dot(Aa, a))
+    dot_x += λ * dot_A[end]
+    push!(as.weights, λ)
+    push!(as.atoms, a)
+    push!(as.dots_x, dot_x)
+    push!(as.dots_A, dot_A)
+    push!(as.dots_b, dot_b)
+    push!(as.weights_prev, λ)
+    push!(as.modified, true)
+    return as
+end
+
+function Base.deleteat!(as::ActiveSetQuadratic, idx::Int)
+    @inbounds for i in 1:idx-1
+        as.dots_x[i] -= as.weights_prev[idx] * as.dots_A[idx][i]
+    end
+    @inbounds for i in idx+1:length(as)
+        as.dots_x[i] -= as.weights_prev[idx] * as.dots_A[i][idx]
+        deleteat!(as.dots_A[i], idx)
+    end
+    deleteat!(as.weights, idx)
+    deleteat!(as.atoms, idx)
+    deleteat!(as.dots_x, idx)
+    deleteat!(as.dots_A, idx)
+    deleteat!(as.dots_b, idx)
+    deleteat!(as.weights_prev, idx)
+    deleteat!(as.modified, idx)
+    return as
+end
+
+function Base.empty!(as::ActiveSetQuadratic)
+    empty!(as.atoms)
+    empty!(as.weights)
+    as.x .= 0
+    empty!(as.dots_x)
+    empty!(as.dots_A)
+    empty!(as.dots_b)
+    empty!(as.weights_prev)
+    empty!(as.modified)
+    return as
+end
+
+function active_set_update!(active_set::ActiveSetQuadratic, lambda, atom, renorm=true, idx=nothing; add_dropped_vertices=false, vertex_storage=nothing)
+    # rescale active set
+    active_set.weights .*= (1 - lambda)
+    active_set.weights_prev .*= (1 - lambda)
+    active_set.dots_x .*= (1 - lambda)
+    # add value for new atom
+    if idx === nothing
+        idx = find_atom(active_set, atom)
+    end
+    if idx > 0
+        @inbounds active_set.weights[idx] += lambda
+        @inbounds active_set.modified[idx] = true
+    else
+        push!(active_set, (lambda, atom))
+    end
+    if renorm
+        add_dropped_vertices = add_dropped_vertices ? vertex_storage !== nothing : add_dropped_vertices
+        active_set_cleanup!(active_set, update=false, add_dropped_vertices=add_dropped_vertices, vertex_storage=vertex_storage)
+        active_set_renormalize!(active_set)
+    end
+    active_set_update_scale!(active_set.x, lambda, atom)
+    return active_set
+end
+
+# only useful for testing purposes, should remain commented out
```

Comment:
we can probably remove it no? Or keep it as a function somewhere in the tests?

Reply:
Sure, this can be removed.